app/template_extractor.py

```diff
@@ -0,0 +1,314 @@
+
+                        self.write_question_translatable_text(question, question_context, msgctxt)
+
+                        for answer in question.get('answers', []):
+                            answer_context = question_context + ['Answer id: ' + answer['id']]
+                            self.write_text_for_container(answer, answer_context, msgctxt)
+                            self.get_guidance_text(answer, 'Answer', answer_context, msgctxt)
+                            self.get_options_text(answer, answer_context, msgctxt)
+                            self.get_validation_text(answer, answer_context, msgctxt)
+
+    def get_title_text_for_context(self, container):
+        if 'titles' in container:
+            return container['titles'][0]['value']
+
+        return container.get('title', '""')
+
+    def write_text_for_container(self, container, context, msgctxt=None):
+
+        if isinstance(container, dict, ):
+            for key in self.keys_for_translation:
+                value = container.get(key)
+                if value is not None and value != '':
+                    value_context = context.copy()
+                    value_context.append('Key: ' + key)
+                    self.add_to_translation_object(value, value_context, msgctxt)
+
+        elif isinstance(container, list):
+            for value in container:
+                self.add_to_translation_object(value, context, msgctxt)
+
+    def add_to_translation_object(self, value, context, msgctxt=None):
+        object_key = (TemplateExtractor.trim_chars_from_object_key(value),
+                      TemplateExtractor.trim_chars_from_object_key(msgctxt))
+        object = {
+            'value': value,
+            'context': context,
+            'msgctxt': msgctxt,
+        }
+
+        if self.translation_object.get(object_key):
+            extra_context = self.translation_object[object_key]['context'] + ['and'] + context
+            object['context'] = extra_context
+
+        self.translation_object[object_key] = object
+
+    @staticmethod
+    def trim_chars_from_object_key(key_value):
+        remove_these_chars = ["'", '“', '”', '‘', '’']
+
+        if key_value:
+            for char in remove_these_chars:
+                key_value = key_value.replace(char, '')
+
+        return  key_value
+
+    def write_to_pot_file(self):
+
+        file = open(self.output_path, 'w')
+        output = self.get_pot_file_header()
+
+        for translation in self.translation_object.values():
+            for context_item in translation['context']:
+                output += '#. {}\n'.format(context_item)
+
+            if translation.get('msgctxt'):
+                output += 'msgctxt "{}"\n'.format(translation['msgctxt'])
+
+            output += 'msgid "{}"\n'.format(translation['value'])
+            output += 'msgstr ""\n\n'
+            # output += 'msgstr "WELSH - {}"\n\n'.format(translation['value'])  # FOR TESTING
+
+
+        file.write(output)
```

Comment:
Getting this when translating census.
```
    self.write_to_pot_file()
  File "/Users/abrahm/Documents/projects/eq-translations/app/template_extractor.py", line 136, in write_to_pot_file
    file.write(output)
UnicodeEncodeError: 'ascii' codec can't encode character '\u2019' in position 8863: ordinal not in range(128)
```

Reply:
As above.

Reply:
Was with census schema from v3. Working with the latest changes.